runtime/src/iree/builtins/ukernel/arch/arm_64/common_arm_neon.h

```diff
@@ -0,0 +1,278 @@
+  int32x4_t v1_i32 = vdupq_n_s32(0);
+  v0_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 0 * stride), v0_i32, 0);
+  v0_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 1 * stride), v0_i32, 1);
+  v0_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 2 * stride), v0_i32, 2);
+  v0_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 3 * stride), v0_i32, 3);
+  v1_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 4 * stride), v1_i32, 0);
+  v1_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 5 * stride), v1_i32, 1);
+  v1_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 6 * stride), v1_i32, 2);
+  v1_i32 =
+      vld1q_lane_s32((const iree_uk_int32_t*)(src + 7 * stride), v1_i32, 3);
+  int8x16x2_t v;
+  v.val[0] = vreinterpretq_s8_s32(v0_i32);
+  v.val[1] = vreinterpretq_s8_s32(v1_i32);
+  return v;
+}
+
+static inline int8x16x4_t iree_uk_neon_load_8x8xi8_strided_permute(
+    const iree_uk_int8_t* src, iree_uk_ssize_t stride, int p0, int p1, int p2,
+    int p3, int p4, int p5, int p6, int p7) {
+  int8x8_t row0 = vld1_s8(src + p0 * stride);
+  int8x8_t row1 = vld1_s8(src + p1 * stride);
+  int8x8_t row2 = vld1_s8(src + p2 * stride);
+  int8x8_t row3 = vld1_s8(src + p3 * stride);
+  int8x8_t row4 = vld1_s8(src + p4 * stride);
+  int8x8_t row5 = vld1_s8(src + p5 * stride);
+  int8x8_t row6 = vld1_s8(src + p6 * stride);
+  int8x8_t row7 = vld1_s8(src + p7 * stride);
+  int8x16x4_t v;
+  v.val[0] = vcombine_s8(row0, row1);
+  v.val[1] = vcombine_s8(row2, row3);
+  v.val[2] = vcombine_s8(row4, row5);
+  v.val[3] = vcombine_s8(row6, row7);
+  return v;
+}
+
+static inline int8x16x4_t iree_uk_neon_load_8x8xi8_strided(
+    const iree_uk_int8_t* src, iree_uk_ssize_t stride) {
+  return iree_uk_neon_load_8x8xi8_strided_permute(src, stride, 0, 1, 2, 3, 4, 5,
+                                                  6, 7);
+}
+
+static inline int16x8x2_t iree_uk_neon_zip_16xi8_as_8xi16(int8x16_t a,
+                                                          int8x16_t b) {
+  int8x16x2_t z = vzipq_s8(a, b);
+  int16x8x2_t r;
+  r.val[0] = vreinterpretq_s16_s8(z.val[0]);
+  r.val[1] = vreinterpretq_s16_s8(z.val[1]);
+  return r;
+}
+
+static inline int32x4x2_t iree_uk_neon_zip_8xi16_as_4xi32(int16x8_t a,
+                                                          int16x8_t b) {
+  int16x8x2_t z = vzipq_s16(a, b);
+  int32x4x2_t r;
+  r.val[0] = vreinterpretq_s32_s16(z.val[0]);
+  r.val[1] = vreinterpretq_s32_s16(z.val[1]);
+  return r;
+}
+
+static inline int64x2x2_t iree_uk_neon_zip_4xi32_as_2xi64(int32x4_t a,
+                                                          int32x4_t b) {
+  int32x4x2_t z = vzipq_s32(a, b);
+  int64x2x2_t r;
+  r.val[0] = vreinterpretq_s64_s32(z.val[0]);
+  r.val[1] = vreinterpretq_s64_s32(z.val[1]);
+  return r;
+}
+
+static inline void iree_uk_neon_copy_8x1xi8_strided_to_unstrided(
+    iree_uk_int8_t* restrict out_ptr, const iree_uk_int8_t* restrict in_ptr,
```

Comment:
This header is only included by C code under this `arm_64` directory.  IIUC, the problem with `restrict` in pure C contexts is that MSVC does not support it, right, and so the reason why this currently work is that we don't currently compile arm_64 code on MSVC? I assume MSVC-compiling-to-arm64 is going to be a thing in the future and that's why we need to switch to `IREE_UK_RESTRICT` everywhere?

Since this is a preexisting issue (I need to fix more occurrences than are in this PR anyway), can I do this in a follow-up (and fix all occurences at once)?

Reply:
yeah msvc for arm64 is what's used for windows-on-arm, so we'll want to make sure this all works - fine to do as a follow-up!

Reply:
Ah didn't see this --- done as part of this PR after all.